### PR TITLE
[TRAFODION-3142] Correct Syntactic Descriptions of Character String Data Types and Add Examples (NCHAR) for *CREATE TABLE Statement* in *Trafodion SQL Reference Manual*

### PR DESCRIPTION
**Pull request by liuyu000**

### CODE REVIEW

File: docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc [hunk 1]
```diff
@@ -2963,18 +2963,22 @@ column-definition is:
 <<<
 ```
 data-type is:
-    char[acter] [(length [characters])]
-          [character set char-set-name]
-          [upshift] [[not]casespecific]
+    char[acter] [length [characters]]
```
Comment (DaveBirdsall) [inline docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc]: I think you meant, "char[acter] (length [characters])"  (That is, the length is surrounded by parentheses, and it is not optional.)

File: docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc [hunk 2]
```diff
@@ -2963,18 +2963,22 @@ column-definition is:
...
-  | varchar (length) [character set char-set-name]
+                        [character set char-set-name]
+                        [upshift] [[not]casespecific]
+  | varchar (length [characters]) 
+            [character set char-set-name]
+            [upshift] [[not]casespecific]
+  | varchar2 (length [characters]) 
+             [character set char-set-name]
+             [upshift] [[not]casespecific]          
+  | nchar [length [characters]]
```
Comment (DaveBirdsall) [inline docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc]: Should be "nchar (length [characters])"

File: docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc [hunk 3]
```diff
@@ -3712,6 +3716,60 @@ ATTRIBUTES ALIGNED FORMAT
 --- SQL operation complete.
 ```
 
+* This example demonstrates the character set of NCHAR column (_c1_) *cannot be specified*.
```
Comment (DaveBirdsall) [inline docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc]: I think one example is probably enough. The point we are trying to make is that "NATIONAL CHAR" implies a particular character set, namely, "UCS2". This is hard-coded into the product; it cannot be changed. Possible word-smith: "NATIONAL CHAR (or NCHAR) is a short-hand for CHAR ... CHARACTER SET UCS2. So it is not possible to specify a CHARACTER SET clause when using NCHAR, as the following example shows:"

File: docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc (hunk 1 above)
Comment (liuyu000) [reply docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc]: Thanks for your review @DaveBirdsall 

`length [characters]` is optional. 
If `length` is not specified, the default is 1.
default

File: docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc (hunk 2 above)
Comment (liuyu000) [reply docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc]: Same as above.
`length [characters]` is optional.

default

File: docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc (hunk 3 above)
Comment (liuyu000) [reply docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc]: Thanks Dave, your comments have been incorporated 😁

File: docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc (hunk 1 above)
Comment (DaveBirdsall) [reply docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc]: Ah, good catch! I stand corrected. So, the original text, "char[acter] [(length [characters])] was correct? (The parentheses are needed if a length is supplied.)

Comment (DaveBirdsall) [reply docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc]: FYI: I double-checked the parser code. Length is optional for CHAR and NCHAR. Length is required for VARCHAR and NCHAR VARYING.

Comment (liuyu000) [reply docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc]: Yes, the origin text is right, I've corrected my writing 😁

File: docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc (hunk 2 above)
Comment (liuyu000) [reply docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc]: Here should be `nchar [(length) [characters]]` 😁

File: docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc [hunk 4]
```diff
@@ -2963,18 +2963,22 @@ column-definition is:
 <<<
 ```
 data-type is:
-    char[acter] [(length [characters])]
-          [character set char-set-name]
-          [upshift] [[not]casespecific]
+    char[acter] [(length) [characters]]
```
Comment (DaveBirdsall) [inline docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc]: Should be "char[acter] [(length [characters])]"  (The trailing parenthesis should come after the optional [characters] syntactic element)

File: docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc [hunk 5]
```diff
@@ -2963,18 +2963,22 @@ column-definition is:
...
-  | varchar (length) [character set char-set-name]
+                        [character set char-set-name]
+                        [upshift] [[not]casespecific]
+  | varchar (length [characters]) 
+            [character set char-set-name]
+            [upshift] [[not]casespecific]
+  | varchar2 (length [characters]) 
+             [character set char-set-name]
+             [upshift] [[not]casespecific]          
+  | nchar [(length) [characters]]
```
Comment (DaveBirdsall) [inline docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc]: Same comment here.

File: docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc (hunk 4 above)
Comment (liuyu000) [reply docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc]: Thanks for your eagle eye 😁 
Sorry it's my careless mistake :pig: I've corrected them all :pencil2:

File: docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc (hunk 5 above)
Comment (liuyu000) [reply docs/sql_reference/src/asciidoc/_chapters/sql_statements.adoc]: Thanks, I've corrected  :rowboat:

**Comment by liuyu000** [comment]
@DaveBirdsall Thanks, yes I can merge this PR if it is approved 😁

**Pull request merged**